### PR TITLE
Changed roles for Nova Compute

### PR DESCRIPTION
**Pull request by adean20csc**
Changed roles for Nova Compute

### CODE REVIEW

File: roles/nova-common/tasks/main.yml [hunk 1]
```diff
@@ -19,4 +19,48 @@
...
+    - "DEFAULT memcache_servers {{ memcached_servers }}"
+    - "DEFAULT rabbit_hosts {{ rabbit_hosts }}"
+    - "DEFAULT rabbit_ha_queues true"
+    - "DEFAULT metadata_host {{ nova_private_vip }}"
+    - "DEFAULT metadata_listen {{ internal_ipaddr }}"
+    - "DEFAULT metadata_listen_port {{ nova_metadata_port }}"
+    - "DEFAULT service_neutron_metadata_proxy True"
+    - "DEFAULT neutron_metadata_proxy_shared_secret {{ neutron_metadata_proxy_shared_secret }}"
+    - "DEFAULT glance_host {{ glance_private_vip }}"
+    - "DEFAULT glance_host {{ glance_port | default(9292) }}"
```
Comment (jcpowermac) [inline roles/nova-common/tasks/main.yml]: Looks like there is two `glance_host` here.

File: roles/nova-compute/tasks/main.yml [hunk 2]
```diff
@@ -50,10 +47,30 @@
...
+- name: modify libvirtd parameters with openstack-config
+  command: openstack-config --verbose --set libvirtd '' LIBVIRTD_ARGS '"--listen"'
+    register: cmd
+    changed_when: "'unchanged' not in cmd.stderr"
+  when: nova_instance_live_migration and openstack_release == 'juno'
   notify: restart libvirtd
   tags: nova
 
+ - name: modify libvirtd.conf parameters with openstack-config
+   command: openstack-config --verbose --set libvirtd.conf {{ item }}
```
Comment (jcpowermac) [inline roles/nova-compute/tasks/main.yml]: Incorrect path for libvirtd.conf

File: roles/nova-compute/tasks/main.yml [hunk 3]
```diff
@@ -103,9 +120,77 @@
...
+    - "DEFAULT rabbit_ha_queues true"
+    - "DEFAULT notification_driver neutron.openstack.common.notifier.rpc_notifier"
+  register: cmd
+  changed_when: "'unchanged' not in cmd.stderr"
+  when: openstack_release == 'juno'
+
+- name: Update ovs_neutron_plugin.ini file for Juno
+  command: openstack-config --verbose --set /etc/neutron//plugins/openvswitch/ovs_neutron_plugin.ini {{ item }}
+  with_items:
+    - "agent tunnel_types vxlan"
```
Comment (jcpowermac) [inline roles/nova-compute/tasks/main.yml]: line 146 - 155 most of this parameters have available variables to use.  See `ovs_neutron_plugini.j2`

File: roles/nova-compute/tasks/main.yml [hunk 4]
```diff
@@ -50,10 +47,30 @@
   template: src={{ openstack_release }}/libvirtd.conf.j2
             dest=/etc/libvirt/libvirtd.conf
             mode=0644 owner=root group=root
-  when: nova_instance_live_migration
+  when: nova_instance_live_migration and openstack_release == 'icehouse'
+  notify: restart libvirtd
+  tags: nova
+
+- name: modify libvirtd parameters with openstack-config
+  command: openstack-config --verbose --set libvirtd '' LIBVIRTD_ARGS '"--listen"'
```
Comment (jcpowermac) [inline roles/nova-compute/tasks/main.yml]: Incorrect path.

**Comment by jcpowermac** [comment]
Looks good for :+1: once travis passes will merge.

File: roles/nova-compute/tasks/main.yml [hunk 5]
```diff
@@ -50,10 +47,30 @@
...
             dest=/etc/libvirt/libvirtd.conf
             mode=0644 owner=root group=root
-  when: nova_instance_live_migration
+  when: nova_instance_live_migration and openstack_release == 'icehouse'
+  notify: restart libvirtd
+  tags: nova
+
+- name: modify libvirtd parameters with openstack-config
+  command: openstack-config --verbose --set /etc/sysconfig/libvirtd '' LIBVIRTD_ARGS '"--listen"'
+    register: cmd
```
Comment (mtnbikenc) [inline roles/nova-compute/tasks/main.yml]: Need to fix the indents on line 56 and 57.

**Comment by mtnbikenc** [comment]
Looks good :+1:

**Pull request merged**